### PR TITLE
SOFTWARE-5924: Remove el7 builds from Github Workflows

### PR DESCRIPTION
**Pull request by mwestphall**
Build 3.6 images in el9 instead of el7

### CODE REVIEW

File: .github/workflows/hotfix-image-builds.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # TODO: detect the appropriate OSG repos and OS for build and installation
 # Currently we assume that upstream's version corresponds to what's in
-# osg-upcoming-el7-build and osg-upcoming-testing
+# osg-upcoming-el9-build and osg-upcoming-testing
```
Comment (mwestphall) [inline .github/workflows/hotfix-image-builds.yml:3]:
```suggestion
# osg-upcoming-el8-build and osg-upcoming-testing
```

File: .github/workflows/hotfix-image-builds.yml [hunk 2]
```diff
@@ -51,7 +51,7 @@ jobs:
               -v $(pwd)/xrootd/:/xrootd \
               -v $(pwd)/_build_dir:/u/_build_dir \
               --privileged \
-              opensciencegrid/osg-build:el7 \
+              opensciencegrid/osg-build:el9 \
```
Comment (mwestphall) [inline .github/workflows/hotfix-image-builds.yml:54]:
```suggestion
              opensciencegrid/osg-build:el8 \
```

File: .github/workflows/release-series-builds.yml [hunk 3]
```diff
@@ -23,7 +23,7 @@ jobs:
         repo: ['development', 'testing', 'release']
         osg_series:
           - name: '3.6'
-            os: 'el7'
+            os: 'el9'
```
Comment (mwestphall) [inline .github/workflows/release-series-builds.yml:26]:
```suggestion
            os: 'el8'
```

File: .github/workflows/release-series-builds.yml [hunk 4]
```diff
@@ -65,7 +65,7 @@ jobs:
...
```
Comment (mwestphall) [inline .github/workflows/release-series-builds.yml:68]:
```suggestion
            os: 'el8'
```

File: .github/workflows/release-series-builds.yml [hunk 5]
```diff
@@ -110,7 +110,7 @@ jobs:
...
```
Comment (mwestphall) [inline .github/workflows/release-series-builds.yml:113]:
```suggestion
            os: 'el8'
```

File: .github/workflows/release-series-builds.yml [hunk 6]
```diff
@@ -180,7 +180,7 @@ jobs:
...
```
Comment (mwestphall) [inline .github/workflows/release-series-builds.yml:183]:
```suggestion
            os: 'el8'
```

**Comment by mwestphall** [comment]
It looks like 3.6/EL9 builds don't work due to xrootd-rucioN2N-for-Xcache not being present in 23-contrib. 3.6/EL8 builds appear to work though.

**Review by matyasselmeci — APPROVED** [review]

**Pull request closed without merging**